### PR TITLE
Version: add RC (release candidate) to the meson version

### PR DESCRIPTION
**Pull request by glimchb**
This is to help uploading python package to pip

see discussion in https://github.com/linux-nvme/libnvme/issues/95 and https://github.com/linux-nvme/libnvme/issues/287

Signed-off-by: Boris Glimcher <Boris.Glimcher@emc.com>

### CODE REVIEW

**Comment by glimchb** [comment]
@igaw Looks like `secrets.TEST_PYPI_API_TOKEN` is empty 

https://github.com/linux-nvme/libnvme/runs/5674662663?check_suite_focus=true

**Comment by igaw** [comment]
Strange, I am pretty sure I added the token. Anyway, created a new one and added it to github.

**Comment by glimchb** [comment]
> Strange, I am pretty sure I added the token. Anyway, created a new one and added it to github.

@igaw, sorry, not working still...

https://github.com/linux-nvme/libnvme/runs/5708205580?check_suite_focus=true
```
Warning:  It looks like you are trying to use an API token to authenticate in the package index and your token value does not start with "pypi-" as it typically should. This may cause an authentication error. Please verify that you have copied your token properly if such an error occurs.
```

**Comment by igaw** [comment]
I think, you don't have the permission to upload. Which make sense that only maintainers are allowed to to trigger the upload.

**Comment by glimchb** [comment]
@igaw I don't think it's about permissions, the message is clear that the secret is not well-defined... 
I can't see the secrets here, but they need to start with  "pypi-" 
we should have 2 secrets `secrets.TEST_PYPI_API_TOKEN` and `secrets.PYPI_API_TOKEN`

**Comment by igaw** [comment]
The token is correctly prefixed. I made sure of that.

**Comment by glimchb** [comment]
let me try again to trigger the upload with verbose mode now

**Comment by glimchb** [comment]
@igaw sorry to persist on this... I see `Environment secrets` vs `Repository secrets`... the secrets are under repository, right ?

**Comment by glimchb** [comment]
thanks @igaw !

**Comment by glimchb** [comment]
- `pyproject.toml` supports dynamic version, so we don't have to change meson version, so this is good.

- but according to https://thiblahute.gitlab.io/mesonpep517/pyproject.html 
```
NOTE: The project version and name are extracted from the meson.build [project()](http://mesonbuild.com/Reference-manual.html#project) section.
```
This is why I'm considering to switch to 
https://meson-python.readthedocs.io/en/stable/usage/start.html

**Comment by glimchb** [comment]
@igaw I just was hoping it would be good if we could actually test if we have the pypi stuff correctly setup and it works before we switch to https://pypi.org/project/meson-python/

**Comment by glimchb** [comment]
closing this once, since @igaw is working on versioning already

**Comment by igaw** [comment]
After ponder on this for while I think we just follow the Linux and model here. We keep the version string until the first rc or proper release. This seems to be an accepted way to handle it.
